pkg/tsdb/postgres/macros.go

```diff
@@ -130,13 +130,19 @@ func (m *postgresMacroEngine) evaluateMacro(name string, args []string) (string,
 				m.query.Model.Set("fillValue", floatVal)
 			}
 		}
-		return fmt.Sprintf("floor(extract(epoch from %s)/%v)*%v", args[0], interval.Seconds(), interval.Seconds()), nil
+
+		if m.query.DataSource.JsonData.Get("timescaledb").MustBool() {
```

Comment:
I would add an argument to newPostgresMacroEngine (https://github.com/grafana/grafana/pull/12680/files#diff-7aea8624475496656ef64377637bbc4dR22) and use an instance variable of the postgresMacroEngine. Will probably make it easier to test as well.

Reply:
Good call, this reduced the patch quite a bit.